core/src/main/java/org/mskcc/cbio/oncokb/util/EvidenceUtils.java

```diff
@@ -1000,4 +997,19 @@ public int compare(Evidence x, Evidence y) {
         }
         return evidences;
     }
+
+    public static Set<Evidence> getAllEvidencesByAlterationsGenes(Collection<Alteration> alterations) {
+        Set<Gene> genes = new HashSet<>();
+        Set<Evidence> evidences = new HashSet<>();
+        for (Alteration alteration : alterations) {
+            genes.add(alteration.getGene());
+        }
+        if (genes.size() == 1) {
```

Comment:
I don't think you need to list this condition separately. It will be handled well in the for loop below.